### PR TITLE
r/virtual_desktop_workspace_application_group_association: fixing import support

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This includes a fix for #9046 - but also spotted a couple of other issues whilst passing through to do fix that

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
This also fixes some bugs in the Data Share and Storage Sync data sources, where errors weren't raised

File: azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource_test.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func testCheckAzureRMDesktopVirtualizationHostPoolDestroy(s *terraform.State) er
 		result, err := client.Get(ctx, id.ResourceGroup, id.Name)
 
 		if err == nil {
-			return fmt.Errorf("Virtual Desktop Host Pool still exists:\n%#v", result)
+			return fmt.Errorf("Virtual Desktop Application Group still exists:\n%#v", result)
```
Comment (jackofallops) [inline azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource_test.go:158]: typo here, revert?

**Review by jackofallops — APPROVED** [review]
Thanks @tombuildsstuff - One really minor item, but LGTM 👍

**Comment by tombuildsstuff** [comment]
Desktop Virtualization tests pass:

Screenshot 2020-11-04 at 17 27 55

**Comment by tombuildsstuff** [comment]
Tests look good 👍

**Pull request merged**